src/main/java/org/threadly/litesockets/TCPClient.java

```diff
@@ -319,4 +322,81 @@ public boolean isEncrypted() {
     }
     throw new IllegalStateException("Must Set the SSLEngine before starting Encryption!");
   }
+  
+  /**
+   * 
+   * @author lwahlmeier
+   *
+   */
+  private class TCPSocketOptions extends BaseClientOptions {
+    
+    @Override
+    public boolean setTcpNoDelay(boolean enabled) {
+        try {
+          channel.socket().setTcpNoDelay(enabled);
+          return true;
```

Comment:
Now I am even more confused about the true/false return.  Above it looked like there were conditions that would always return false.  But here it looks like a retry may work.

I definitely think we need to clean this up somehow.  If we want the true/false to be checked and retried by the implementor, maybe when it is not possible to set the value we should just throw an `UnsupportedOperationException`

Further more, I wonder if there is a way we can retry on behalf of the implementor?  I am not sure what this would look like, possibly returning a future till it's set?  If we do it async, we then have to consider conflicting configuration coming in parallel though.